### PR TITLE
Should interrupt sessions for flow run contact when user delete the run

### PR DESCRIPTION
**Pull request by norkans7**
Issue here is that user delete a run, then the contact send a new message and the flow reply again so the contact is kept in the flow since we were not interrupting the sessions
This PR fixes that

### CODE REVIEW

File: temba/flows/models.py [hunk 1]
```diff
@@ -1939,6 +1939,9 @@ def release(self, delete_reason=None):
             for recent in FlowPathRecentRun.objects.filter(run=self):
                 recent.release()
 
+            if self.is_active or delete_reason == FlowRun.DELETE_FOR_USER:
+                mailroom.queue_interrupt(self.org, contacts=[self.contact])
```
Comment (nicpottier) [inline temba/flows/models.py]: Also, this seems a bit broad.. this is interrupting all flows for the contact, which feels risky, especially as you could be looking at a run that is in the past.

Comment (nicpottier) [reply temba/flows/models.py]: Seems like the right thing here is probably to pass in the run session id.

**Comment by norkans7** [comment]
I can reproduce the issue locally, I now understood a little more on the mailroom tasks
I added the session id instead of run id

**Comment by norkans7** [comment]
So we queue the task only if the session has the status of WAITING only
since even changing EXPIRED or FAILED to INTERRUPTED seems not right either

right @nicpottier ?

**Comment by nicpottier** [comment]
I think that's right yes.

File: temba/flows/models.py [hunk 2]
```diff
@@ -1939,6 +1939,9 @@ def release(self, delete_reason=None):
             for recent in FlowPathRecentRun.objects.filter(run=self):
                 recent.release()
 
+            if delete_reason == FlowRun.DELETE_FOR_USER and self.session.status == FlowSession.STATUS_WAITING:
```
Comment (nicpottier) [inline temba/flows/models.py]: Session is NULLABLE and will be null for any run that has been completed for 7 days, so need to check that it isn't none before firing this too. (good news is that sessions are only trimmed after completion so there's nothing to do in that case)

File: temba/flows/tests.py [hunk 3]
```diff
@@ -3908,29 +3908,51 @@ def _check_deletion(self, delete_reason, expected):
...
+                .visit(beer_split)
+                .wait()
+                .resume(msg=self.create_incoming_msg(self.contact, "primus"))
+                .set_result("Beer", "primus", "Primus", "primus")
+                .visit(name_prompt, exit_index=2)
+                .send_msg("Mmmmm... delicious Turbo King. Lastly, what is your name?")
+                .visit(name_split)
+                .wait()
+                .save()
+            )
```
Comment (rowanseymour) [inline temba/flows/tests.py]: btw you could avoid a bit of repetition here by not repeating the bits that are common to both paths

**Pull request merged**